Cleanup

gwenchee opened the pull request:
In this PR, I removed functionality and files that are no longer used in ROLLO. 

Previously, I intended for ROLLO to have in-built functions for OpenMC and Moltres. However, upon further reflection, I think ROLLO will be more usable in the long run if was nuclear-code agnostic. All of these removals are associated with `openmc_evaluation` and `moltres_evaluation`. I also removed OpenMC dependencies from the unit tests. 

I also previously had special variables that users could define. This is no longer used. All of these removals are associated with `SpecialVariables`

pep8speaks commented:
Hello @gwenchee! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

































* In the file [`tests/unit_tests/input_test_files/input_test_eval_fn_generator_openmc_template.py`](https://github.com/arfc/rollo/blob/6301557878033ed6295612d524db02ce8134de57/tests/unit_tests/input_test_files/input_test_eval_fn_generator_openmc_template.py):

> [Line 1:32](https://github.com/arfc/rollo/blob/6301557878033ed6295612d524db02ce8134de57/tests/unit_tests/input_test_files/input_test_eval_fn_generator_openmc_template.py#L1): [W292](https://duckduckgo.com/?q=pep8%20W292) no newline at end of file

yardasol reviewed (approved):
- [x] Read the PR description
- [x] Read through all the changes, considering the following questions.
  - [x] Is there anything you can praise about this PR? Start with
        praise.
  - [x] Are variable names brief but descriptive?
  - [x] Are new/changed functions no longer than a paragraph?
  - [x] Do all function parameters have default values where appropriate?
  - [x] Is the code clear and clean? (see Robert C. Martin's
        [Clean Code](https://i-share-uiu.primo.exlibrisgroup.com/discovery/fulldisplay?docid=alma99944155312205899&context=L&vid=01CARLI_UIU:CARLI_UIU&tab=LibraryCatalog&lang=en))
  - [x] Is there enough documentation?
  - [ ] Does the programming style meet the requirements of the
        repository ([PEP8](https://www.python.org/dev/peps/pep-0008/) for python,
        [google C++ style guide](https://google.github.io/styleguide/cppguide.html), etc.)
  - [x] If the repository has continuous integration: does the PR pass
        the tests?
  - [x] Is the PR free of random cruft (built files, `.swp` files,
        etc.)?
  - [x] Do all files in the PR belong in the repository?
  - [x] If the PR deletes files, is this appropriate?
  - [x] Make a review, leaving kind comments and suggesting changes
        where needed (to resolve the above).
  - [ ] Has the author resolved all of the comments and suggestions
        in your review?
- [ ] When you approve of the PR, merge and close it.
- [x] Thank the author for their contribution.

Hi @gwenchee, thanks for your contribution. Pretty clean removals. I have a few clarifying questions:
1. You remove the majority of the lines from `tests/unit_tests/input_test_files/input_test_eval_fn_generator_openmc_output.py` and `tests/unit_tests/input_test_files/input_test_eval_fn_generator_openmc_template.py`, but you don't outright delete these files. If you are making ROLLO code-agnostic, why do you need to keep these code-specific files?
2. You remove lines related to `polynomial_triso` in several files. Can you explain what the significance of this is?

Going to approve for now.

gwenchee commented on tests/unit_tests/input_test_files/input_test_eval_fn_generator_openmc_output.py:
```diff
@@ -1,5 +1 @@
-sp = openmc.StatePoint("statepoint.10.h5")
-num_batches = sp.n_batches
-print({"num_batches": num_batches})
+print({"num_batches": 10})
```
Previously, I had OpenMC code in my unit tests to test its functionality with ROLLO. But, now I'm going in the direction of nuclear code agnostic ROLLO functionality. And so I updated this test to be a hardcoded value. The hardcoded value still enables the test to fill the purpose of ROLLO reading the output value

yardasol commented on rollo/input_validation.py:
```diff
@@ -395,22 +395,8 @@ def validate_evaluators(self, input_evaluators):
             evaluators sub-dictionary from input file
 
         """
-        # evaluators available
```
You're removing this block to make Rollo code agnostic, yes?

gwenchee replied:
Yeah!

yardasol commented on tests/unit_tests/test_evaluation.py:
```diff
@@ -31,34 +31,34 @@ def test_eval_fn_generator():
     )
     ev.add_evaluator(
-        solver_name="moltres",
+        solver_name="evaluator_2",
```
evaluator 2 is a reactor physics code?

yardasol commented on tests/unit_tests/test_evaluation.py:
```diff
@@ -31,34 +31,34 @@ def test_eval_fn_generator():
     ev = Evaluation()
     ev.add_evaluator(
-        solver_name="openmc",
+        solver_name="evaluator_1",
```
Evaluator 1 is a monte carlo particle transport code?

yardasol reviewed (approved):
Thanks for explaining!

Going off of your new changes, I take it that `evaluator1` is a monte carlo particle transport code and `evaluator2` is a reactor physics code?

Also looks like your changes brought up some PEP8 issues.

gwenchee commented:
I replaced openmc to be evaluator_1 and moltres to be evaluator_2. Technically, they can be any reactor modeling software. They do not have to be monte carlo and reactor physics software. Having evaluator_1 and evaluator_2 tests that you can have multiple evaluators in ROLLO input file. 

Also, I addressed the pep8 issues @yardasol

gwenchee commented:
@yardasol bump
